src/main/java/com/microsoft/jenkins/keyvault/BaseSecretCredentials.java

```diff
@@ -65,18 +67,35 @@ void setSecretGetter(SecretGetter secretGetter) {
     }
 
     interface SecretGetter {
-        SecretBundle getKeyVaultSecret(String pCredentialId, String aSecretIdentifier);
+        KeyVaultSecret getKeyVaultSecret(String pCredentialId, String aSecretIdentifier);
 
+        int NAME_POSITION = 2;
+        int VERSION_POSITION = 3;
         SecretGetter DEFAULT = new SecretGetter() {
 
             @Override
-            public SecretBundle getKeyVaultSecret(String pCredentialId, String aSecretIdentifier) {
-                KeyVaultCredentials keyVaultCredentials = AzureCredentials.getCredentialById(pCredentialId);
-                final KeyVaultClient client = new KeyVaultClient(keyVaultCredentials);
-
-                SecretBundle secret = client.getSecret(aSecretIdentifier);
-                client.httpClient().connectionPool().evictAll();
-                return secret;
+            public KeyVaultSecret getKeyVaultSecret(String pCredentialId, String aSecretIdentifier) {
```

Comment:
this is the breaking change for AzureKeyVault, but as long as releases are co-ordinated it will be fine